### PR TITLE
JS Capstone project

### PR DESCRIPTION
**Pull request by tajemouti**
## Hi Reviewer 👋 ##
Here's a short [video presentation](https://drive.google.com/file/d/1c2QcuqCZrvGFSkRXu_mb_0TOkoOCpBrl/view?usp=sharing) and a list of the things we did working on this project:

1. [Implemented HTML/CSS/JS linters](https://github.com/tajemouti/js-capstone-project/commit/e320bb77842c632375a7150c85c66fb70546439a).
2. [Implemented Webpack boiterplate](https://github.com/tajemouti/js-capstone-project/commit/ce7104685df96cfb939063793f88517d7e8d0938).
3. [Implemented the test environment](https://github.com/tajemouti/js-capstone-project/commit/3952b3e2d26094c7c9f6c8c793a0637ecabfb940).
4. [Created a header with a navigation bar](https://github.com/tajemouti/js-capstone-project/commit/7750273f6bebf2dce7570607cc3f94f7b4fb63be).
5. [Added a logo and some basic styling](https://github.com/tajemouti/js-capstone-project/commit/f939a0caa30cf2722664b598c03233d60676e3e8).
6. [Added a footer section](https://github.com/tajemouti/js-capstone-project/commit/45a4fd58c2af02d915a49fcbbbd35d2c2665e54c).
7. [Added basic styling to the footer](https://github.com/tajemouti/js-capstone-project/commit/ed45a1cc9dfd00c328b24a8dde92989a871d84db).
8. [Added homepage and modals sections](https://github.com/tajemouti/js-capstone-project/commit/24d508f8d4c0f073db298d4637eee5ebf4c5467d).
9. [Displayed meals by category on the homepage](https://github.com/tajemouti/js-capstone-project/commit/25d70d3e2ed813cb28ebc319d06d5bda6c6ec81c).
10. [Added comments popups with API integration](https://github.com/tajemouti/js-capstone-project/commit/7162183a17e35fc2bf5d731e3e8349aa291690af).
11. [Added count along with comments list](https://github.com/tajemouti/js-capstone-project/commit/501bb000d9d3b6537af486f8cd17087038b0b16d).
12. [Added the Likes functionality ](https://github.com/tajemouti/js-capstone-project/commit/a1acc5d14463e6b8c2108c472e3604d37d686ee6).
13. [Added unit tests to the counter functions](https://github.com/tajemouti/js-capstone-project/commit/858d1171c8b5b1f3f97cb0f38c2ed24839c18fcf).
14. [Deployed the project on GitHub Pages](https://tajemouti.github.io/js-capstone-project/dist/).
15. [Record a presentation video on Zoom](https://drive.google.com/file/d/1-FwCiTOHKV4ZsV4Qn1ZfrypW0-0Da7RG/view?usp=sharing).
16. Edited the [README](https://github.com/tajemouti/js-capstone-project/blob/dev/README.md) file.
17. [Fixed linters errors](https://github.com/tajemouti/js-capstone-project/commit/299f262fffac11dda404095200f0da848b782e3f).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+mealListCont.addEventListener('click', async (e) => {
+  if (e.target.classList.contains('likes-icon')) {
+    const mealTitleElement = e.target.closest('.meal').querySelector('h3');
+    const mealTitle = mealTitleElement.textContent;
+    const likesCountElement = e.target.nextElementSibling;
+    const currentLikes = parseInt(likesCountElement.textContent, 10);
+    const updatedLikes = currentLikes + 1;
+    const success = await sendLike(mealTitle);
+    if (success) {
+      updateLikesCount(likesCountElement, updatedLikes);
```
Comment (t-yanick) [inline src/index.js:41]: - [ ] Great job working on this capstone project. As per project requirements, you are supposed to used the Involvement API to show the item likes which I don't think is the case at the moment. You are only using the Involvement API at the moment for the comments counter. Kindly make use of this API for item likes.
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @tajemouti and @BilalLiaquat7 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Use of correct gitflow :+1: 
:heavy_check_mark: Effective use of ES6 and Webpack
:heavy_check_mark: Use of Jest for counters tests
:heavy_check_mark: Interfaces look exactly like given mockups
:heavy_check_mark: Popup for comments well implemented
:heavy_check_mark: App works as expected

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (tajemouti) [reply src/index.js:41]: ## Hi @t-yanick ##
Thank you for the fast and insightful review. I really appreciate it.
Actually, we used the Involvement API to render the likes to the meals cards, the displayMeals function in the render.js file. This function is called when rendering each meal on the page, and it involves fetching the number of likes for each meal from the Involvement API and updating the corresponding likes count element in the HTML.

- Here's the relevant part of the displayMeals function:

1. `const likes = await fetchLikes(meal.strMeal);`:

This line fetches the number of likes for the current meal using the fetchLikes function, passing the meal title (meal.strMeal) as an argument.

2. `updateLikesCount(likesCount, likes);`:
 
This line calls the updateLikesCount function, which takes the likes count element (likesCount) and the fetched likes value (likes) as arguments. The purpose of this call is to update the displayed likes count for the current meal in the HTML.

- Now, allow me to kindly walk you through this piece of code line by line:

mealListCont.addEventListener('click', async (e) => {
  if (e.target.classList.contains('likes-icon')) {
    const mealTitleElement = e.target.closest('.meal').querySelector('h3');
    const mealTitle = mealTitleElement.textContent;
    const likesCountElement = e.target.nextElementSibling;
    const currentLikes = parseInt(likesCountElement.textContent, 10);
    const updatedLikes = currentLikes + 1;
    const success = await sendLike(mealTitle);
    if (success) {
      updateLikesCount(likesCountElement, updatedLikes);
    }
  }
});

This piece of code is responsible for handling the user interaction when the "Like" button (heart icon) for a specific meal in the web application is clicked. Let's break down what each part of the code is doing:

1. `mealListCont.addEventListener('click', async (e) => { ... })`:
   - This adds a click event listener to the `mealListCont` element, which is the container for displaying meals on the webpage.
   - The listener is set up to react to click events on any element inside the `mealListCont` container.

2. `if (e.target.classList.contains('likes-icon')) { ... }`:
   - This conditional statement checks whether the clicked element has a class of `'likes-icon'`. It's used to specifically target the heart icon (like button) that was clicked.

3. `const mealTitleElement = e.target.closest('.meal').querySelector('h3');`:
   - This line finds the closest ancestor element with the class `'meal'` from the clicked element, then selects the `<h3>` element within that ancestor.
   - This `<h3>` element contains the title of the meal.

4. `const mealTitle = mealTitleElement.textContent;`:
   - This extracts the text content (title) of the selected meal's `<h3>` element.

5. `const likesCountElement = e.target.nextElementSibling;`:
   - This selects the next sibling element of the clicked element. In this case, it's the element that displays the number of likes for the corresponding meal.

6. `const currentLikes = parseInt(likesCountElement.textContent, 10);`:
   - This converts the text content of the element that displays the current number of likes into an integer. This allows us to perform numerical calculations.

7. `const updatedLikes = currentLikes + 1;`:
   - This calculates the updated number of likes after the user clicks the like button.

8. `const success = await sendLike(mealTitle);`:
   - This invokes the `sendLike` function with the `mealTitle` as an argument.
   - The `sendLike` function sends a request to the Involvement API to record the user's like for the specific meal.
   - The `await` keyword ensures that the function waits for the API request to complete and returns a success status.

9. `if (success) { ... }`:
   - If the `sendLike` function call was successful (meaning the API request was successful), this condition is met.

10. `updateLikesCount(likesCountElement, updatedLikes);`:
    - This calls the `updateLikesCount` function to update the displayed likes count on the webpage.
    - The function takes the `likesCountElement` (the DOM element displaying likes count) and the `updatedLikes` value as arguments.

In summary, this code listens for clicks on the "Like" button for a meal, then retrieves the meal title and current likes count. After that, it sends a like to the Involvement API, and if the API request is successful, it updates the displayed likes count on the webpage.

- Finally, here's a live [demo](https://drive.google.com/file/d/1EBDBThRFdnC1hiNZ_u6E1tUoU5vGxyeZ/view?usp=sharing) that shows clearly the interaction of the Likes functionality with the Involvement API.

Thank you for taking the time to go through all of this.
Cheers

**Comment by t-yanick** [comment]
Thank you for the explicit explanations.  Kindly submit for re-review
normally.  Thanks and regards

File: README.md [hunk 2]
```diff
@@ -1 +1,158 @@
...
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank Microverse.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](/LICENSE) license.
```
Comment (Felix45) [inline README.md:156]: - [ Optional ] I think it would be nice if you fix the License link on your readme file, I noticed that when I click on the link it leads to a `404` page not found error, a license serves to inform anyone who would like to use your project how it is Licensed. I would like to suggest that you add a file called `LICENSE` to the root directory of your project to fix this. You can use the content on this [link >> ](https://opensource.org/license/mit/) as the body of your License, remember to customize it.

## Error on link
Screenshot from 2023-08-12 00-11-24

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @tajemouti  & @BilalLiaquat7, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Application works as expected :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Application has passing tests :white_check_mark:    
- [x] Implemented all the required changes from the previous reviewer :white_check_mark:   

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
